src/debug/providerRegistry.ts

```diff
@@ -0,0 +1,41 @@
+import * as vscode from "vscode";
+
+import { IDebugProvider, IDockerParser } from "./debugInterfaces";
+import { JavaDebugProvider } from "./javaDebugProvider";
+import { JavaDockerResolver } from "./javaDockerResolver";
+
+const supportedProviders: IDebugProvider[] = [
+    new JavaDebugProvider(new JavaDockerResolver())
+];
+
+async function showProviderPick(): Promise<IDebugProvider> {
+    if (supportedProviders.length < 1) {
+        return null;
+    } else if (supportedProviders.length === 1) {
+        return supportedProviders[0];
+    }
+
+    const providerItems = supportedProviders.map((provider) => {
+        return {
+            label: provider.getDebuggerType(),
+            provider
+        };
+    });
+    
+    const pickedProvider = await vscode.window.showQuickPick(<any[]> providerItems, { placeHolder: "Please select the debug type." });

```

Comment:
I don't know what 'debug type' means?

Reply:
It's similar as `Select Environment`. In vscode, when your workspace doesn't contain `launch.json` file, and click F5 will ask user to select a environment first.

I would change the tooltip message to `Select Environment`.